src/Expectation.php

```diff
@@ -177,6 +177,56 @@ public function sequence(...$callbacks): Expectation
         return $this;
     }
 
+    /**
+     * If the subject matches one of the expressions, the callback in the expression is run.
+     *
+     * @template TMatchValue
+     *
+     * @param Closure|bool|string                            $subject
+     * @param array<mixed, callable(self): void|TMatchValue> $expressions
+     *
+     * @return \Pest\Expectation
+     */
+    public function match($subject, array $expressions): Expectation
+    {
+        $subject = is_callable($subject)
+            ? $subject
+            : function () use ($subject) {
+                return $subject;
+            };
+
+        $subject   = $subject();
+        $keys      = array_keys($expressions);
+        $matched   = false;
+
+        if (in_array($subject, ['0', '1', false, true], true)) {
+            $subject = (int) $subject;
+        }
+
+        foreach ($expressions as $key => $callback) {
+            if ($subject !== $key) {
+                continue;
+            }
+
+            $matched = true;
+
+            if (is_callable($callback)) {
+                $callback(new self($this->value));
+                continue;
+            }
+
+            (new self($this->value))->toEqual($callback);
+
+            break;
+        }
+
+        if (!$matched) {
+            test()->addWarning('No item found matching "' . $subject . '".');
```

Comment:
This should be removed.

Reply:
And add a `$this->toHaveKey(` before all this code.

Reply:
@nunomaduro Can you explain this in more detail please? I don't understand where to add it. `$this->toHaveKey(..?.)`